subtests/docker_cli/systemd/systemd.py

```diff
@@ -0,0 +1,71 @@
+"""
+Summary
+-------
+
+Operational Summary
+---------------------
+    1. Copy Unit file
+    2. Build image using predefined Dockerfile & p4321-server.py
+    3. Run image
+    4. Confirm p4321-server script is running
+    5. Stop container
+    6. cleanup
+
+"""
+
+import shutil
+import time
+from os.path import join
+from autotest.client import utils
+from autotest.client.shared.utils import is_port_free
+from dockertest.subtest import Subtest
+from dockertest.dockercmd import DockerCmd
+
+
+class systemd(Subtest):
+
+    def initialize(self):
+        super(systemd, self).initialize()
+        sysd_unitf = join(self.bindir, 'p4321.service')
+        shutil.copyfile(sysd_unitf, self.config['sysd_unitf_dest'])
+        for build_file in ['Dockerfile', 'p4321-server.py']:
+            shutil.copy(join(self.bindir, build_file), self.tmpdir)
+        self.build_image()
+
+    def build_image(self):
+        dkrcmd = DockerCmd(self, 'build', ['--force-rm -t',
+                                           self.config['name'], self.tmpdir])
+        dkrcmd.execute()
+        time.sleep(5)
+
+    def run_once(self):
+        self.sysd_action(action='daemon-reload')
+        self.sysd_action(action='start',
+                         sysd_unit=self.config['sysd_unit_file'])
+        while is_port_free(self.config['tcp_port'], self.config['tcp_ip']):
+            time.sleep(2)
+        # FIXME
+        # for now, using cat but
+        # will try something similar to below with socket
+        # read_port = open('/dev/tcp/localhost/4321').read().strip()
+        system_time = utils.run("cat < /dev/tcp/localhost/4321",
+                                ignore_status=False)
+        system_time_adigit = system_time.stdout.strip().isdigit()
+        self.failif(not system_time_adigit,
+                    "Returned system time is not a digit")
+        self.sysd_action(action='stop',
+                         sysd_unit=self.config['sysd_unit_file'])
+
+    @staticmethod
+    def sysd_action(action=None, sysd_unit=None):
+        command = "systemctl " + action
+        if sysd_unit:
+            command += " " + sysd_unit
+        utils.run(command, ignore_status=False)
+        time.sleep(2)
+
+    def cleanup(self):
+        super(systemd, self).cleanup()
+        utils.run("rm -f %s" % self.config['sysd_unitf_dest'])
```

Comment:
``os.unlink()`` is the python-way.  Though it can be annoying if the file isn't there.   This works well:

```python
import errno
...
filepath = self.config['sysd_unitf_dest']
try:
    os.unlink(filepath)
except OSError, xcpt:
    if xcpt.errno != errno.ENOENT:  # No such file or directory
        raise
    self.logwarning("Failed to remove %s, it's completely invisible!"
                    % filepath)
```